felix/dataplane/linux/bpf_route_mgr.go

```diff
@@ -230,6 +246,11 @@ func (m *bpfRouteManager) calculateRoute(cidr ip.V4CIDR) *routes.Value {
 		flags |= routes.FlagHost
 	}
 
+	if m.dsrOptoutCIDRs.Contains(cidr) {
```

Comment:
`For this feature I think you need to check if the route is inside any of the `dsrOptoutCIDRs` For (bad) example, if the user sets `DSROptout=0.0.0.0/0` then all traffic should be excluded, even if it's also a node IP, or something from an IP pool.  The logic will put 0.0.0.0/0 in the routes map but the IP pool routes will be excluded.

I think we've got a trie implementation in the ip package, should be trivial to use the trie instead of the set and then the contains check will just do the right thing.

Reply:
Yep, `CIDRTrie.Covers` does the check

Reply:
OK, excellent, I was pondering that and then decided that it is meant for external cidrs only. This will make it robust :+1: